docs/blog/_posts/2020-12-18-scala3-m3.md

```diff
@@ -0,0 +1,209 @@
+---
+layout: blog-page
+title: Scala 3.0.0-M3: developer's preview before RC1
+author: Anatolii Kmetiuk
+authorImg: /images/anatolii.png
+date: 2020-12-18
+---
+We are happy to announce the release of Scala 3.0.0-M3. This release is the Developer's Preview release intended to contain all the features meant for RC1 which is tentatively planned for January 2021. The purpose of M3 is to give the community a chance to try out all the features and give us feedback before sealing them in RC1.
+
+To collect the feedback, we have designed a [Scala 3 Developer’s Preview satisfaction survey (3-5 min)](https://docs.google.com/forms/d/e/1FAIpQLSflVmTu9lhrtnSTh2tKAjUGrt3WvEgwlDqZg66O3EVSXd1aJg/viewform?usp=sf_link) – please fill it in to express how you feel about Scala 3.
+
+For the weeks between M2 and M3, we were tying up all the loose ends so that to end 2020 having done everything intended for the release. Also, we have used this time as the last chance to review and fine-tune the language syntax. A large number of consultations with the community took place during this time so that to ensure everyone is on the same page.
+
+In this article, you will find the summary of the most important changes that took place.
+
+<!--more-->
+# SBT plugin bump
+SBT plugin was bumped to 0.5.0. Because of the changes in PR [#10607](https://github.com/lampepfl/dotty/pull/10607) this release of Scala 3 will not work with earlier versions of the SBT plugin. So if you are an early adopter of Scala 3 who has some projects in it, you will need to bump your SBT plugin version to migrate to M3.
+
+# Final syntactic tweaks
+## `as` dropped from the `given` syntax
+The following syntax is obsolete:
+
+```scala
+given intOrd as Ordering[Int]:
+  ...
+given listOrd[T: Ordering] as Ordering[List[T]]:
+  ...
+
+given Ordering[Int]:
+  ...
+given [T: Ordering] as Ordering[List[T]]:
+  ...
+
+given global as ExecutionContext = ForkJoinContext()
+given Context = ctx
+```
+
+Here is how the above is expressed now:
+
+```scala
+given intOrd: Ordering[Int] with
+  ...
+given listOrd[T: Ordering]: Ordering[List[T]] with
+  ...
+
+given Ordering[Int] with
+  ...
+given [T: Ordering]: Ordering[List[T]] with
+  ...
+
+given global: ExecutionContext = ForkJoinContext()
+given Context = ctx
+```
+
+You can find an extensive discussion on the above change in its [PR #10538](https://github.com/lampepfl/dotty/pull/10538).
+
+## Drop `as` in patterns
+Since we dropped `as` from `given`s, we lost a strong reason for having `as` at all. For this reason, we dropped them also from patterns. If previously the correct binding syntax is patterns was:
```

Comment:
```suggestion
Since we dropped `as` from `given`s, we lost a strong reason for having `as` at all. Therefore, we dropped `as` from patterns as well. The following syntax, valid in Scala 3.0.0-M2, is not accepted anymore:
```